packages/openfeature-js/src/lib/client.ts

```diff
@@ -29,49 +26,64 @@ export class OpenFeatureClient implements Feature {
     this._trace = trace.getTracer(OpenFeatureClient.name);
   }
 
-  // TODO: see if a default callback makes sense here
-  async isEnabled(id: string, context?: Context): Promise<boolean> {
-    return (await this.evaluateFlag('is_enabled', id, context)).enabled;
+  isEnabled(flagId: string, defaultValue: boolean, context?: Context): Promise<boolean> {
+    return this.getBooleanValue(flagId, defaultValue, context);
   }
 
-  async getVariation(
-    id: string,
-    context?: Context
-  ): Promise<FlagEvaluationVariationResponse> {
-    return this.evaluateFlag('variation', id, context);
+  getBooleanValue(flagId: string, defaultValue: boolean, context?: Context): Promise<boolean> {
+    return this.evaluateFlag('boolean', flagId, defaultValue, context);
   }
 
-  private async evaluateFlag(
-    type: 'is_enabled' | 'variation',
+  getStringValue(flagId: string, defaultValue: string, context?: Context): Promise<string> {
+    return this.evaluateFlag('string', flagId, defaultValue, context);
+  }
+
+  getNumberValue(flagId: string, defaultValue: number, context?: Context): Promise<number> {
+    return this.evaluateFlag('number', flagId, defaultValue, context);
+  }
+
+  getObjectValue<T extends object>(flagId: string, defaultValue: T, context?: Context): Promise<T> {
+    return this.evaluateFlag('json', flagId, defaultValue, context);
+  }
+
+  private async evaluateFlag<T extends boolean | string | number | object>(
+    type: FlagType,
     id: string,
+    defaultValue: T,
     context?: Context
-  ) {
+  ): Promise<T> {
     const span = this.startSpan(`feature flag - ${type}`);
     try {
       span.setAttribute(SpanProperties.FEATURE_FLAG_ID, id);
 
-      const response = await this.getProvider().evaluateFlag({
-        flagId: id,
-        context: context ?? {},
-        clientName: this._name,
-        clientVersion: this._version,
-      });
-
-      span.setAttribute(SpanProperties.FEATURE_FLAG_ENABLED, response.enabled);
-
-      if (response.stringValue) {
-        span.setAttribute(
-          SpanProperties.FEATURE_FLAG_VARIATION_STRING,
-          response.stringValue
-        );
+      const provider = this.getProvider();
+      let valuePromise: boolean | string | number | object;
```

Comment:
Perhaps we should change to the `object` type to something like this:

```typescript
type JSONValue =
    | string
    | number
    | boolean
    | { [x: string]: JSONValue }
    | Array<JSONValue>;
```
> NOTE: Requires TypeScript 3.7 and newer.

Reply:
As discussed, for now we'll use object, we don't want to support JSON values such as `1` or `true`.